src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkAppointmentLinker.kt

```diff
@@ -0,0 +1,116 @@
+package uk.gov.justice.digital.hmpps.whereabouts.services
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.whereabouts.model.PrisonAppointment
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkAppointment
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBooking
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkAppointmentRepository
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkBookingRepository
+import javax.transaction.Transactional
+
+typealias AppointmentFinder = (PrisonAppointment?) -> VideoLinkAppointment?
+
+@Service
+class VideoLinkAppointmentLinker(
+  private val videoLinkAppointmentRepository: VideoLinkAppointmentRepository,
+  private val videoLinkBookingRepository: VideoLinkBookingRepository,
+  private val prisonApiService: PrisonApiService
+) {
+
+  fun linkAppointments() {
```

Comment:
This rate at which this runs will be limited by the speed at which prison-api can return appointments per bookingId.    There are ~ 4300 bookingIds for which to fetch and filter all appointments since 2019-01-01.  There's no throttling here - perhaps there should be?

Reply:
Yeah it would be good to batch the calls as requesting appointments are a bit slow sometimes.